evm/p2p/peer.py

```diff
@@ -449,6 +452,23 @@ def les_proto(self):
         return self._les_proto
 
 
+class ETHPeer(BasePeer):
+    _eth_proto = None
+    _supported_sub_protocols = [ETHProtocol]
+
+    @property
+    def eth_proto(self):
+        assert len(self.enabled_sub_protocols) > 0
+        if self._eth_proto is None:
+            for proto in self.enabled_sub_protocols:
+                if proto.name == ETHProtocol.name:
+                    self._eth_proto = proto
```

Comment:
It feels like this loop is missing a `break` statement.  I *think* that there's an implicit understanding that there aren't multiple values in `self.enabled_sub_protocols` which have the same name.  However, it bothers me that this code *can* set `self._eth_proto` multiple times (albeit to the same value) before exiting.

If we add a break, we can also change the next line from being `if self._eth_proto is None` to be an `else` clause  for this `for` loop which I think is *slightly* more semantically correct for how this code flows.

* Loop over proto names and set the local `_eth_proto`
* Break when you find the proper protocol
* If the loop never breaks, then an error has occurred.

Reply:
There can be multiple values with the same name in `.enabled_sub_protocols`, but they will have different versions, and we actually want to continue in the loop until the last protocol that matches `ETHProtocol.name` because that's going to be the one with the highest version. This is all a bit of a hack, really; we need a generic mechanism to access the protocols supported by a given Peer instance, but I wanted to wait until we have more use cases before attempting to implement it.
Would it be ok with you if I just added a FIXME comment here for now?

Reply:
>  Would it be ok with you if I just added a FIXME comment here for now?

Yes, along with the explanation about there being multiple values and why it's written this way.